samples/snippets/csharp/how-to/strings/CompareStrings.cs

```diff
@@ -0,0 +1,255 @@
+            // </Snippet2>
+        }
+
+        private static void OrdinalStaticComparison()
+        {
+            // <Snippet3>
+            string root = @"C:\users";
+            string root2 = @"C:\Users";
+
+            bool areEqual = String.Equals(root, root2, StringComparison.Ordinal);
+            Console.WriteLine($"Ordinal static: <{root}> and <{root2}> are {(areEqual ? "equal." : "not equal.")}");
+            areEqual = String.Equals(root, root2, StringComparison.OrdinalIgnoreCase);
+            Console.WriteLine($"Ordinal staticignore case: <{root}> and <{root2}> are {(areEqual ? "equal." : "not equal.")}");
+            // </Snippet3>
+        }
+        private static void ReferenceEqualAndInterning()
+        {
+            // <Snippet4>
+            string a = "The computer ate my source code.";
+            string b = "The computer ate my source code.";
+
+            if (String.ReferenceEquals(a, b))
+                Console.WriteLine("a and b are interned.");
+            else
+                Console.WriteLine("a and b are not interned.");
+
+            string c = String.Copy(a);
+
+            if (String.ReferenceEquals(a, c))
+                Console.WriteLine("a and c are interned.");
+            else
+                Console.WriteLine("a and c are not interned.");
+            // </Snippet4>
+        }
+
+        private static void SortArrayOfStrings()
+        {
+            // <Snippet5>
+            string[] lines = new string[]
+            {
+                @"c:\public\textfile.txt",
+                @"c:\public\textFile.TXT",
+                @"c:\public\Text.txt",
+                @"c:\public\testfile2.txt"
+            };
+
+            Console.WriteLine("Non-sorted order:");
+            foreach (string s in lines)
+            {
+                Console.WriteLine($"   {s}");
+            }
+
+            Console.WriteLine("\n\rSorted order:");
+
+            // Specify Ordinal to demonstrate the different behavior.
+            Array.Sort(lines, StringComparer.Ordinal);
+
+            foreach (string s in lines)
+            {
+                Console.WriteLine($"   {s}");
+            }
+            // </Snippet5>
+        }
+
+        private static void SearchSortedArray()
+        {
+            // <Snippet6>
+            string[] lines = new string[]
+            {
+                @"c:\public\textfile.txt",
+                @"c:\public\textFile.TXT",
+                @"c:\public\Text.txt",
+                @"c:\public\testfile2.txt"
+            };
+            Array.Sort(lines, StringComparer.Ordinal);
+
+            string searchString = @"c:\public\TEXTFILE.TXT";
+            Console.WriteLine($"Binary search for <{searchString}>");
+            int result = Array.BinarySearch(lines, searchString, StringComparer.OrdinalIgnoreCase);
+            ShowWhere<string>(lines, result);
+
+            Console.WriteLine($"{(result > 0 ? "Found" : "Did not find")} {searchString}");
```

Comment:
See my comment from the text.
Rather than showing the search string if it's found, it would be clearer to display the found string.